common/testing/taskpoller/taskpoller.go

```diff
@@ -47,29 +48,47 @@ type (
 		namespace string
 	}
 	Options struct {
-		tv                  *testvars.TestVars
-		timeout             time.Duration
-		pollStickyTaskQueue bool
+		tv                      *testvars.TestVars
+		timeout                 time.Duration
+		pollActivityTaskRequest *workflowservice.PollActivityTaskQueueRequest
+		pollworkflowTaskRequest *workflowservice.PollWorkflowTaskQueueRequest
 	}
 	// OptionFunc is a function to change an Options instance
 	OptionFunc func(*Options)
 )
 
 var (
-	// DrainWorkflowTask returns an empty RespondWorkflowTaskCompletedRequest response
+	// DrainWorkflowTask returns an empty RespondWorkflowTaskCompletedRequest
 	DrainWorkflowTask = func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondWorkflowTaskCompletedRequest, error) {
 		return &workflowservice.RespondWorkflowTaskCompletedRequest{}, nil
 	}
+	// CompleteActivityTask returns a RespondActivityTaskCompletedRequest with an auto-generated `Result` from `tv.Any().Payloads()`.
+	CompleteActivityTask = func(tv *testvars.TestVars) func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error) {
+		return func(task *workflowservice.PollActivityTaskQueueResponse) (*workflowservice.RespondActivityTaskCompletedRequest, error) {
+			return &workflowservice.RespondActivityTaskCompletedRequest{
+				Result: tv.Any().Payloads(),
+			}, nil
+		}
+	}
+	// WithActivityTaskPollRequest defines the PollActivityTaskQueueRequest to use for polling
+	WithActivityTaskPollRequest = func(req *workflowservice.PollActivityTaskQueueRequest) OptionFunc {
+		return func(o *Options) {
+			o.pollActivityTaskRequest = req
+		}
+	}
+	// WithWorkflowTaskPollRequest defines the PollWorkflowTaskQueueRequest to use for polling
+	WithWorkflowTaskPollRequest = func(req *workflowservice.PollWorkflowTaskQueueRequest) OptionFunc {
+		return func(o *Options) {
+			o.pollworkflowTaskRequest = req
+		}
+	}
```

Comment:
I considered not using OptionFunc here and instead make it a require argument, but it seems like this will only be used for special reasons (like worker versioning).